substrate/client/network/common/src/sync.rs

```diff
@@ -379,7 +386,8 @@ pub trait ChainSync<Block: BlockT>: Send {
 	/// Call when a peer has disconnected.
 	/// Canceled obsolete block request may result in some blocks being ready for
 	/// import, so this functions checks for such blocks and returns them.
-	fn peer_disconnected(&mut self, who: &PeerId);
+	#[must_use]
```

Comment:
Why only for this function?

Reply:
Done, added for other functions where appropriate.